crates/red_knot_python_semantic/resources/mdtest/shadowing/function.md

```diff
@@ -25,3 +25,29 @@ def f(): ...
 
 f: int = 1
 ```
+
+## Explicit shadowing involving `def` statements
+
+Since a `def` statement is a declaration, one `def` can shadow another `def`, or shadow a previous
+non-`def` declaration, without error.
+
+```py path=a.py
```

Comment:
Explicit path isn't required, and we usually don't include it unless it is relevant to the test; default filename is `test.py`.

```suggestion
```py
```